Black-Market/Black-Market/Components/MarketTexfield.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  MarketTexfield.swift
+//  Black-Market
+//
+//  Created by Junior Sancho on 7/12/23.
+//
+
+import SwiftUI
+
+struct MarketTexfield: TextFieldStyle {
+    @Binding var style: Style
+    private let title: String?
+        
+    enum Style {
+        case password
+        case `default`
+        case search
+        
+        var icon: Image? {
+            switch self {
+            case .password:
+                return Image("visibilityOffIcon")
+            case .search:
+                return Image("searchIcon")
+            default:
+                return nil
+            }
+        }
+    }
+    
+    init(
+        style: Binding<Style> = .constant(.default),
+        title: String? = nil
+    ) {
+        self._style = style
+        self.title = title
+    }
+    
+    func _body(configuration: TextField<Self._Label>) -> some View {
```

Comment:
Just out of curiosity, why the underscore?. Remember, bindings are represented with an underscore, for example, _quantity, so we have to be cautious with its use.

Reply:
It's actually possible to create our own style via a "hidden" _body method, therefore we could think about the actual TextFieldStyle declaration as following:
```
public protocol TextFieldStyle {
  associatedtype _Body: View
  @ViewBuilder func _body(configuration: TextField<Self._Label>) -> Self._Body
  typealias _Label = _TextFieldStyleLabel
}
```
...making it possible to create our own styles:

```
struct CustomTextFieldStyle: TextFieldStyle {
  func _body(configuration: TextField<_Label>) -> some View {
     //
  }
}
```